### PR TITLE
Add wait for delete test

### PR DESCRIPTION
**Pull request by benjaminjb**
**Checklist:**
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [ ] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**
 - [ ] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [x] Testing enhancement
 - [ ] Other

**What is the current behavior (link to any open issues here)?**

Our delete KUTTL test wasn't waiting for the pods to delete before comparing delete events.

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Now we wait, also grab the replica time first since that should happen first.

**Other Information**:

### CODE REVIEW

File: testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml [hunk 1]
```diff
@@ -25,16 +25,20 @@ commands:
       if [ -z "$REPLICA" ]; then exit 1; fi
 
       kubectl delete postgrescluster -n "${NAMESPACE}" delete-switchover
+      
+      kubectl wait pod/"${REPLICA}" --namespace "${NAMESPACE}" --for=delete --timeout=-1s
```
Comment (cbandy) [inline testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml:29]: ⛏️ 
```suggestion
      kubectl wait "pod/${REPLICA}" --namespace "${NAMESPACE}" --for=delete --timeout=-1s
```

Comment (cbandy) [inline testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml:29]: ❓ `--timeout=-1s` means one week. Can we do anything sooner, like `5m` or `1h`?

🤔 How does this interact with the KUTTL timeout (~3m iirc?)

File: testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml [hunk 2]
```diff
@@ -25,16 +25,20 @@ commands:
...
+      KILLING_REPLICA_TIMESTAMP=$(
         kubectl get events --namespace="${NAMESPACE}" \
-          --field-selector reason="Killing",involvedObject.fieldPath="spec.containers{database}",involvedObject.name="${PRIMARY}" \
+          --field-selector reason="Killing",involvedObject.fieldPath="spec.containers{database}",involvedObject.name="${REPLICA}" \
           --output=jsonpath={.items..firstTimestamp}
       )
-      
-      KILLING_REPLICA_TIMESTAMP=$(
+
+      kubectl wait pod/"${PRIMARY}" --namespace "${NAMESPACE}" --for=delete --timeout=-1s
```
Comment (cbandy) [inline testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml:37]: 🤔 What does this command do if the pod is already gone?

**Review by cbandy — APPROVED** [review]

File: testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml (hunk 2 above)
Comment (benjaminjb) [reply testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml:37]: `wait --for=delete` seems to have three outcomes: waiting (until timeout), observing the delete (returns "condition met"), failing to find the object (returns immediately, no error):
```➜  ~ k wait pod/task-pv-pod --for=delete
error: timed out waiting for the condition on pods/task-pv-pod
➜  ~ k wait pod/task-pv-pod --for=delete
pod/task-pv-pod condition met
➜  ~ k wait pod/task-pv-pod --for=delete
➜  ~ ```

So seems safe to wait for a non-existent object to be deleted, but open to discussing and maybe going over the code.

File: testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml (hunk 1 above)
Comment (benjaminjb) [reply testing/kuttl/e2e/delete/13-delete-cluster-and-check.yaml:29]: Just ran a test and a sleep greater than the timeout does cause the test to error with a timeout. So no matter what we set this timeout to, if it's over 180 seconds, KUTTL will fail.

**Review by cbandy — APPROVED** [review]

**Pull request merged**